homeassistant/components/bring/todo.py

```diff
@@ -0,0 +1,177 @@
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import BringData, BringDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor from a config entry created in the integrations UI."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    unique_id = config_entry.unique_id
+
+    if TYPE_CHECKING:
+        assert unique_id
+
+    async_add_entities(
+        BringTodoListEntity(
+            coordinator,
+            bringList=bringList,
+            unique_id=unique_id,
+        )
+        for bringList in coordinator.data.values()
+    )
+
+
+class BringTodoListEntity(
+    CoordinatorEntity[BringDataUpdateCoordinator], TodoListEntity
+):
+    """A To-do List representation of the Bring! Shopping List."""
+
+    _attr_icon = "mdi:cart"
+    _attr_has_entity_name = True
+    _attr_supported_features = (
+        TodoListEntityFeature.CREATE_TODO_ITEM
+        | TodoListEntityFeature.UPDATE_TODO_ITEM
+        | TodoListEntityFeature.DELETE_TODO_ITEM
+        | TodoListEntityFeature.SET_DESCRIPTION_ON_ITEM
+    )
+
+    def __init__(
+        self,
+        coordinator: BringDataUpdateCoordinator,
+        bringList: BringData,
+        unique_id: str,
+    ) -> None:
+        """Initialize BringTodoListEntity."""
+        super().__init__(coordinator)
+        self._list_uuid = bringList["listUuid"]
+        self._attr_name = bringList["name"]
+        self._attr_unique_id = f"{unique_id}_{self._list_uuid}"
+
+    @property
+    def todo_items(self) -> list[TodoItem]:
+        """Return the todo items."""
+        return [
+            TodoItem(
+                uid=item["name"],
+                summary=item["name"],
+                description=item["specification"] or "",
+                status=TodoItemStatus.NEEDS_ACTION,
+            )
+            for item in self.bringList["items"]
+        ]
+
+    @property
+    def bringList(self) -> BringData:
+        """Return the bring list."""
+        return self.coordinator.data[self._list_uuid]
+
+    async def async_create_todo_item(self, item: TodoItem) -> None:
+        """Add an item to the To-do list."""
+        try:
+            await self.hass.async_add_executor_job(
+                self.coordinator.bring.saveItem,
+                self.bringList["listUuid"],
+                item.summary,
+                item.description or "",
+            )
+        except BringRequestException as e:
+            _LOGGER.warning("Unable to save todo item for bring")
+            raise HomeAssistantError from e
```

Comment:
Please also fix the others like we did with the other one

Reply:
This is the last one :)